mergify_engine/actions/post_check.py

```diff
@@ -86,13 +87,16 @@ def _post(self, ctxt, rule, missing_conditions):
                 self.config["summary"], extra_variables
             )
         except context.RenderTemplateFailure as rmf:
-            return (
-                "failure",
+            return check_api.Result(
+                check_api.Conclusion.FAILURE,
                 "Invalid summary template",
                 str(rmf),
             )
 
-        return ("failure" if missing_conditions else "success", title, summary)
+        if missing_conditions:
+            return check_api.Result(check_api.Conclusion.FAILURE, title, summary)
+        else:
+            return check_api.Result(check_api.Conclusion.SUCCESS, title, summary)
```

Comment:
This seems to be outside the scope of this patch?

Reply:
Hum, it's the same code as before with the new enum/dataclass

Reply:
My bad I misread the original code with the hidden `if`